feat(gemini): add support for image generation output from Gemini models

minguncle opened the pull request:
## Issue
Closes  #3635

## Change
Added support for receiving generated images from Gemini's image generation models (e.g., gemini-2.5-flash-image-preview).

**Changes made:**
- Modified `PartsAndContentsMapper.fromGPartsToAiMessage()` to process `inlineData` containing generated image data
- Added `GeneratedImageHelper` utility class for easy access to generated images from `AiMessage.attributes()`
- Generated images are stored in `AiMessage.attributes()` under key "generated_images" 
- Solution works for both streaming and non-streaming responses
- Maintains full backward compatibility with existing code

**Technical details:**
- Processes Gemini response parts with `inlineData` and image mime types
- Converts base64 image data to `Image` objects
- Uses existing `attributes` mechanism to avoid breaking changes
- Includes comprehensive test coverage

This enables full Gemini image generation support while preserving the existing architecture.

## General checklist
- [X] There are no breaking changes
- [X] I have added unit and/or integration tests for my change
- [X] The tests cover both positive and negative cases
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [X] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green
- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

minguncle committed:
feat(gemini): add support for image generation output from Gemini models

minguncle committed:
style(gemini): apply Spotless code formatting

dliubarskyi commented on langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/PartsAndContentsMapper.java:
```diff
@@ -205,99 +203,119 @@ static AiMessage fromGPartsToAiMessage(List<GeminiPart> parts,
         String thinking = thoughts.stream().collect(joining("\n\n"));
         String thinkingSignature = thoughtSignatures.stream().collect(joining("\n\n"));
 
+        // Build attributes map with thinking signature and generated images
```
```suggestion
```

dliubarskyi commented on langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/PartsAndContentsMapper.java:
```diff
@@ -205,99 +203,119 @@ static AiMessage fromGPartsToAiMessage(List<GeminiPart> parts,
                 functionCalls.add(part.getFunctionCall());
             }
+
+            // Handle generated images from inlineData
```
```suggestion
```

dliubarskyi committed:
Update langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/PartsAndContentsMapper.java

dliubarskyi committed:
Update langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/PartsAndContentsMapper.java

dliubarskyi reviewed (commented):
@minguncle please add tests in dev.langchain4j.model.googleai.GoogleAiGeminiChatModelIT and dev.langchain4j.model.googleai.GoogleAiGeminiStreamingChatModelIT

dliubarskyi commented on langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/GeneratedImageHelper.java:
```diff
@@ -0,0 +1,50 @@
+     * @return a list of generated images, or empty list if none found
+     */
+    @SuppressWarnings("unchecked")
+    public static List<Image> getGeneratedImages(AiMessage aiMessage) {
```
Can Gemini generate multiple images in one request?

minguncle replied:
the image generation API supports multiple images via numberOfImages parameter.

  For content generation API (what this PR addresses), while Google's docs don't mention multi-image generation, the response structure supports it - multiple Part objects can contain inlineData. I designed the method to return List<Image> to be future-proof, even though currently only one image is
  typically generated per request.

dliubarskyi commented:
Please also document this feature in docs/docs/integrations/language-models/google-ai-gemini.md

minguncle committed:
test(gemini): add test for image generation output

minguncle committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/PartsAndContentsMapper.java
#	langchain4j-google-ai-gemini/src/test/java/dev/langchain4j/model/googleai/PartsAndContentsMapperTest.java

minguncle committed:
style(gemini): apply Spotless code formatting

minguncle committed:
docs(gemini): add image generation output documentation

minguncle commented:
@dliubarskyi All done! I've added the integration tests for both GoogleAiGeminiChatModelIT and GoogleAiGeminiStreamingChatModelIT, and updated the documentation in docs/docs/integrations/language-models/google-ai-gemini.md as requested.

dliubarskyi commented:
@minguncle thank you! But I can't find `GoogleAiGeminiChatModelIT` and `GoogleAiGeminiStreamingChatModelIT` changes in this PR, perhaps you forgot to push them?

minguncle committed:
test(gemini): add test for image generation output for both GoogleAiGeminiChatModelIT and GoogleAiGeminiStreamingChatModelIT

minguncle committed:
style(gemini): apply Spotless code formatting

dliubarskyi committed:
Merge branch 'main' into feat-gemini-support-image-generation

dliubarskyi reviewed (approved):
@minguncle thanks a lot, good job!

@glaforge would you like to review as well?
